src/Analyzer/QueryNode.h

```diff
@@ -556,10 +556,13 @@ class QueryNode final : public IQueryTreeNode
     }
 
     /// Resolve query node projection columns
-    void resolveProjectionColumns(NamesAndTypes projection_columns_value)
-    {
-        projection_columns = std::move(projection_columns_value);
-    }
+    void resolveProjectionColumns(NamesAndTypes projection_columns_value);
```

Comment:
I will add refactoring of `QueryNode` and `UnionNode` to have `QueryNodeBase` parent class and extract these methods and other common methods into this parent class.

Reply:
We need to do the same with `TableNode` and `TableFunctionNode`